drivers/accel/adxl367/adxl367.c

```diff
@@ -0,0 +1,1310 @@
+ * @param enable - 1 - ENABLE
+ * 				   2 - DISABLE
+ *
+ * @return 0 in case of success, negative error code otherwise.
+*******************************************************************************/
+int32_t adxl367_temp_read_en(struct adxl367_dev *dev, uint8_t enable)
+{
+	return adxl367_reg_write_msk(dev,
+				     ADXL367_REG_TEMP_CTL,
+				     enable ? ADXL367_TEMP_EN : 0U,
+				     ADXL367_TEMP_EN);
+}
+
+/***************************************************************************//**
+ * @brief Enables ADC reading. Disables temperature reading.
+ *
+ * @param dev    - The device structure.
+ * @param enable - 1 - ENABLE
+ * 				   2 - DISABLE
+ *
+ * @return 0 in case of success, negative error code otherwise.
+*******************************************************************************/
+int32_t adxl367_adc_read_en(struct adxl367_dev *dev, uint8_t enable)
+{
+	int32_t ret;
+
+	//disable temp reading
+	ret = adxl367_reg_write_msk(dev,
+				    ADXL367_REG_TEMP_CTL,
+				    0U,
+				    ADXL367_TEMP_EN);
+	if (ret)
+		return ret;
+
+	return adxl367_reg_write_msk(dev,
+				     ADXL367_REG_ADC_CTL,
+				     enable ? ADXL367_ADC_EN : 0U,
+				     ADXL367_ADC_EN);
+}
+
+/***************************************************************************//**
+ * @brief Reads the raw temperature of the device. If ADXL367_TEMP_EN is not
+ *  set, use adxl367_temp_read_en() first to enable temperature reading.
+ *
+ * @param dev      - The device structure.
+ * @param raw_temp - Raw value of temperature.
+ *
+ * @return 0 in case of success, negative error code otherwise.
+*******************************************************************************/
+int32_t adxl367_read_raw_temp(struct adxl367_dev *dev, int16_t *raw_temp)
+{
+	int32_t ret;
+	uint8_t temp[2] = {0, 0};
+	uint8_t reg_data, nready = 1U;
+
+	while (nready) {
+		ret = adxl367_get_register_value(dev, &reg_data, ADXL367_REG_STATUS, 1);
+		if (ret)
+			return -1;
+		if (reg_data & ADXL367_STATUS_DATA_RDY)
+			nready = 0U;
+	}
+
+	ret = adxl367_get_register_value(dev,
+					 temp,
+					 ADXL367_REG_TEMP_H,
+					 2);
+	if (ret)
+		return ret;
+
+	*raw_temp = ((int16_t)temp[0] << 6) + (temp[1] >> 2);
+	// extend sign to 16 bits
+	if (*raw_temp & NO_OS_BIT(13))
+		*raw_temp |= NO_OS_GENMASK(15, 14);
+
+	return 0;
+}
+
+/***************************************************************************//**
+ * @brief Converts raw temperature value to Celsius degree value.
+ *
+ * @param dev       - The device structure.
+ * @param raw_temp  - Raw temperature value.
+ * @param temp   	- Struct to store converted data.
+ *
+ * @return 0 in case of success, negative error code otherwise.
+*******************************************************************************/
+static int32_t adxl367_temp_conv(struct adxl367_dev *dev,
+				 uint16_t raw_temp, struct adxl367_fractional_val* temp)
+{
+	int64_t temp_data;
+
+	temp_data = (int16_t)raw_temp;
+
+	temp_data += ADXL367_TEMP_OFFSET;
```

Comment:
can condense to: `temp_data = (temp_data + ADXL367_TEMP_OFFSET) * ADXL367_TEMP_SCALE`